src/torchmetrics/functional/image/scc.py

```diff
@@ -0,0 +1,177 @@
+from torchmetrics.utilities.distributed import reduce
+
+
+def _scc_update(preds: Tensor, target: Tensor, hp_filter: Tensor, window_size: int) -> Tuple[Tensor, Tensor, Tensor]:
+    """Update and returns variables required to compute Spatial Correlation Coefficient.
+
+    Args:
+        preds: Predicted tensor
+        target: Ground truth tensor
+        hp_filter: High-pass filter tensor
+        window_size: Local window size integer
+
+    Return:
+        Tuple of (preds, target, hp_filter) tensors
+
+    Raises:
+        ValueError:
+            If ``preds`` and ``target`` have different number of channels
+            If ``preds`` and ``target`` have different shapes
+            If ``preds`` and ``target`` have invalid shapes
+            If ``window_size`` is not a positive integer
+            If ``window_size`` is greater than the size of the image
+
+    """
+    if preds.dtype != target.dtype:
+        target = target.to(preds.dtype)
+    _check_same_shape(preds, target)
+    if len(preds.shape) not in (3, 4):
+        raise ValueError(
+            "Expected `preds` and `target` to have batch of colored images with BxCxHxW shape"
+            "  or batch of grayscale images of BxHxW shape."
+            f" Got preds: {preds.shape} and target: {target.shape}."
+        )
+
+    if len(preds.shape) == 3:
+        preds = preds.unsqueeze(1)
+        target = target.unsqueeze(1)
+
+    if not window_size > 0:
+        raise ValueError(f"Expected `window_size` to be a positive integer. Got {window_size}.")
+
+    if window_size > preds.size(2) or window_size > preds.size(3):
+        raise ValueError(
+            f"Expected `window_size` to be less than or equal to the size of the image."
+            f" Got window_size: {window_size} and image size: {preds.size(2)}x{preds.size(3)}."
+        )
+
+    preds = preds.to(torch.float32)
+    target = target.to(torch.float32)
+    hp_filter = hp_filter[None, None, :].to(dtype=preds.dtype, device=preds.device)
+    return preds, target, hp_filter
+
+
+def _symmetric_reflect_pad_2d(input: Tensor, pad: Union[int, Tuple[int, ...]]) -> Tensor:
+    if isinstance(pad, int):
+        pad = (pad, pad, pad, pad)
+    assert len(pad) == 4
+
+    left_pad = input[:, :, :, 0 : pad[0]].flip(dims=[3])
+    right_pad = input[:, :, :, -pad[1] :].flip(dims=[3])
+    padded = torch.cat([left_pad, input, right_pad], dim=3)
+
+    top_pad = padded[:, :, 0 : pad[2], :].flip(dims=[2])
+    bottom_pad = padded[:, :, -pad[3] :, :].flip(dims=[2])
+    return torch.cat([top_pad, padded, bottom_pad], dim=2)
+
+
+def _signal_convolve_2d(input: Tensor, kernel: Tensor) -> Tensor:
+    left_padding = int(torch.floor(tensor((kernel.size(3) - 1) / 2)).item())
+    right_padding = int(torch.ceil(tensor((kernel.size(3) - 1) / 2)).item())
+    top_padding = int(torch.floor(tensor((kernel.size(2) - 1) / 2)).item())
+    bottom_padding = int(torch.ceil(tensor((kernel.size(2) - 1) / 2)).item())
```

Comment:
add `import math` add the top.
Should be faster as we do not have to convert to tensor and back again
```suggestion
    left_padding = int(math.floor((kernel.size(3) - 1) / 2))
    right_padding = int(math.ceil((kernel.size(3) - 1) / 2))
    top_padding = int(math.floor((kernel.size(2) - 1) / 2))
    bottom_padding = int(math.ceil((kernel.size(2) - 1) / 2))
```